src/three-components/Model.js

```diff
@@ -49,10 +49,24 @@ export default class Model extends Object3D {
   }
 
   applyEnvironmentMap(map) {
+    let materialStack = [];
     this.modelContainer.traverse(obj => {
-      if (obj && obj.isMesh && obj.material) {
-        obj.material.envMap = map;
-        obj.material.needsUpdate = true;
+      if (obj && obj.isMesh && Array.isArray(obj.material)) {
+        materialStack = materialStack.concat(obj.material);
+      }
+      else if (obj && obj.isMesh && obj.material) {
+        materialStack.push(obj.material);
+      }
+
+      for (let material of materialStack) {
```

Comment:
Did you mean to do iteration of the material stack on every invocation of the `traverse` callback?

Reply:
Mm, yes, but missing clearing it out. I don't like any way that attempts to not duplicate the logic for the (untested) edge case as well. Will change